impl/client/src/main/javascript/web/util/_a11y.js

```diff
@@ -35,6 +35,358 @@ define(["./_focus"], function(focusUtil) {
     delete: 46
   };
 
+  /*
+   * This software or document includes material copied from or derived from https://www.w3.org/WAI/content-assets/wai-aria-practices/patterns/listbox/examples/js/listbox.js
+   * Copyright © 2023 World Wide Web Consortium. https://www.w3.org/Consortium/Legal/2023/doc-license
+   *
+   */
+  function makeAccessibleListbox(elem) {
+    elem.setAttribute("role", "listbox");
+    elem.setAttribute("tabindex", "0");
+    elem.setAttribute("aria-activedescendant", "");
+    elem.setAttribute("aria-multiselectable", "true");
+    var activeDescendantId = null;
+    var startRangeIndex = 0;
+    var keysSoFar = "";
+    var clearKeysSoFarHandle = null;
+    var handleFocusChange = function () {};
+
+    registerEvents();
+
+    /**
+     *  Register events for the listbox interactions
+     */
+    function registerEvents() {
+      elem.addEventListener("keydown", onKeyPress);
+      elem.addEventListener("click", onClick);
+    }
+
+    function getActiveOption() {
+      return document.getElementById(activeDescendantId);
+    }
+
+    /**
+     * Check if an item is clicked on. If so, focus on it and select it.
+     *
+     * @param evt - The click event object
+     */
+    function onClick(evt) {
+      if (evt.target.getAttribute("role") !== "option") {
+        return;
+      }
+
+      setActiveOption(evt.target);
+      toggleSelectItem(evt.target);
+      updateScroll();
+
+      if (evt.shiftKey) {
+        selectRange(startRangeIndex, evt.target);
+      }
+    }
+
+    /**
+     * Handle various keyboard controls; UP/DOWN will shift focus; SPACE selects an item.
+     *
+     * @param evt - The keydown event object
+     */
+    function onKeyPress(evt) {
+      var key = evt.which || evt.keyCode;
+      var lastActiveId = activeDescendantId;
+      activeDescendantId = elem.getAttribute("aria-activedescendant");
+      var allOptions = getAllOptions();
```

Comment:
Variable not needed. Can use `getAllOptions()` instead inline in the `end` and `A` cases that use it.